### PR TITLE
Sync api keys with a redis instance

### PR DESCRIPTION
**Pull request by stevenday**
Basic functionality to sync api keys in the Django database with a redis
database that Varnish can look at to manage requests.
- SETs an appropriate key whenever the APIKey is saved
- UNSETs the key when the APIKey is deleted

### CODE REVIEW

File: api_keys/tests.py [hunk 1]
```diff
@@ -51,3 +75,51 @@ def test_prints_api_key(self):
...
+        key.delete()
+
+    @override_settings(REDIS_API_NAME='test_api')
+    def test_deletes_key_from_redis(self):
+        expected_key = "key:test_key:api:test_api"
+        r = redis_connection()
+        key = APIKey.objects.create(user=self.user, key="test_key")
+        self.assertEqual(r.get(expected_key), '1')
+        key.delete()
+        self.assertIsNone(r.get(expected_key), '1')
```
Comment (dracos) [inline api_keys/tests.py]: I doubt you want `1` as the test message here :)

File: api_keys/tests.py [hunk 2]
```diff
@@ -51,3 +75,51 @@ def test_prints_api_key(self):
...
+    @override_settings(REDIS_API_NAME='test_api')
+    def test_deleted_users_cascade(self):
+        expected_key = "key:test_key:api:test_api"
+        key = APIKey.objects.create(user=self.user, key="test_key")
+        r = redis_connection()
+        self.assertEqual(r.get(expected_key), '1')
+        self.user.delete()
+        with self.assertRaises(APIKey.DoesNotExist):
+            APIKey.objects.get(pk=key.pk)
+        self.assertIsNone(r.get(expected_key), '1')
```
Comment (dracos) [inline api_keys/tests.py]: Ditto

File: mapit-gb/settings.py [hunk 3]
```diff
@@ -144,12 +144,18 @@
 )
 
 # List of callables that know how to import templates from various sources.
-TEMPLATE_LOADERS = (
-    ('django.template.loaders.cached.Loader', (
+if DEBUG:
+    TEMPLATE_LOADERS = (
```
Comment (dracos) [inline mapit-gb/settings.py]: Unlikely to ever matter here, but in general you could set TEMPLATE_LOADERS and then if not DEBUG wrap it in the cached loader (so you don't have to duplicate the lower level loaders).

**Comment by dracos** [comment]
Looks good, I might squash it up a bit, but no biggie.

**Comment by stevenday** [comment]
Merged into master in e4dcab286d50a7f3e0f4f70f78d1519cec106693

**Pull request closed without merging**